context/src/main/java/io/grpc/Context.java

```diff
@@ -316,23 +365,49 @@ public CancellableContext withDeadline(Deadline deadline,
    *
    */
   public <V> Context withValue(Key<V> k1, V v1) {
-    return new Context(this, keyValueEntries.put(k1, v1));
+    PersistentHashArrayMappedTrie<Key<?>, Object> newKeyValueEntries = keyValueEntries.put(k1, v1);
+    if (shouldSample()) {
+      withValueDupeCounts.addAndGet(
```

Comment:
It's not obvious what withValueDupeCounts is tracking, can we add a comment?

Reply:
Done.

Reply:
I think it should be `N - oldSize + newSize` for this stat to work, like if we added a key and the size increased from 3 to 4:

1 + 4 - 3 = 2 (right now we report 2 duplicates)
Instead if it's flipped we get:
1 + 3 - 4 = 0 (report 0 duplicates)